src/slice-assign.c

```diff
@@ -27,6 +27,11 @@ r_obj* vec_assign_opts(r_obj* x,
                                        KEEP(vec_names(x)),
                                        &location_opts));
 
+  if (vec_size(index) == 0) {
+    FREE(2);
+    return x;
+  }
+
   // Cast and recycle `value`
   value = KEEP(vec_cast(value, x, opts.value_arg, opts.x_arg, opts.call));
```

Comment:
I think the cast and recycle checks need to run before we can think about early exiting